lib/components/bm-confirm-location-on-map.js

```diff
@@ -0,0 +1,148 @@
+/* @flow */
+'use strict'
+
+const geolocation = require('@blinkmobile/geolocation')
+
+/* :: import type {
+  AngularCallbackOptions, Coordinates, NgModelController
+} from '../../types.js' */
+
+const mod = require('../module.js')
+const utils = require('../utils.js')
+
+class BmConfirmLocationOnMapController {
+  /* :: static $inject : string[] */
+
+  // internal use
+
+  /* :: $digest: () => void */
+  /* :: coords: ?Coordinates */
+  /* :: isEditing: boolean */
+
+  // public attributes (after casts / checks)
+
+  /* :: ngDisabled: boolean */
+  /* :: ngModel: NgModelController */
+  /* :: ngReadonly: boolean */
+
+  constructor ($scope /* : Object */) {
+    this.$digest = $scope.$digest.bind($scope)
+    this.coords = null
+    this.isEditing = false
+  }
+
+  $onInit () {}
+
+  $onChanges () {
+    this.ngDisabled = utils.parseBooleanAttribute(this.ngDisabled)
+    this.ngReadonly = utils.parseBooleanAttribute(this.ngReadonly)
+    this.coords = this.ngModel.$viewValue
+  }
+
+  onCancel (event /* : Event */) {
+    this.coords = this.ngModel.$viewValue
+    this.isEditing = false
+  }
+
+  onChange (value /* : ?Coordinates */) {
+    this.coords = value
+  }
+
+  onClear (event /* : Event */) {
+    this.coords = null
+    this.ngModel.$setViewValue(this.coords, event)
+  }
+
+  onConfirm (event /* : Event */) {
+    this.isEditing = false
+    this.ngModel.$setViewValue(this.coords, event)
```

Comment:
I like that the way you've handled roll back and committing of the value to the model, but it may not be needed

There's a [ngModelController](https://docs.angularjs.org/api/ng/type/ngModel.NgModelController) which has `$rollbackViewValue`, `$commitViewValue` etc which is kinda new and made to address this problem. IIRC because you have `require: 'ngModel'` what you actually get is the ngModelController.

also theres [ngModelOptions](https://docs.angularjs.org/api/ng/directive/ngModelOptions) - it has a way of specifying how model updates are done and in theory a dev could attach it to the element and it changes how the model is updated

Although I have found working with the ngModelController to be pretty painful in the past.

I dont think this is a blocker though, because we'll be controlling the html via forms for now. I also think I should have a look at my camera wrapper and check this!!